File: backend/app/services/countries.py
"""Country metadata helpers for workspace creation."""

from datetime import datetime

COUNTRY_FLAGS: dict[str, str] = {
    "Japan": "🇯🇵",
    "Italy": "🇮🇹",
    "China": "🇨🇳",
    "France": "🇫🇷",
    "Spain": "🇪🇸",
    "Germany": "🇩🇪",
    "United Kingdom": "🇬🇧",
    "United States": "🇺🇸",
    "South Korea": "🇰🇷",
    "Thailand": "🇹🇭",
    "Vietnam": "🇻🇳",
    "Indonesia": "🇮🇩",
    "Turkey": "🇹🇷",
    "Greece": "🇬🇷",
    "Portugal": "🇵🇹",
    "Netherlands": "🇳🇱",
    "Switzerland": "🇨🇭",
    "Austria": "🇦🇹",
    "Czech Republic": "🇨🇿",
    "Poland": "🇵🇱",
    "Mexico": "🇲🇽",
    "Brazil": "🇧🇷",
    "Argentina": "🇦🇷",
    "Morocco": "🇲🇦",
    "Egypt": "🇪🇬",
    "UAE": "🇦🇪",
    "Singapore": "🇸🇬",
    "India": "🇮🇳",
    "Australia": "🇦🇺",
    "Canada": "🇨🇦",
    "Norway": "🇳🇴",
    "Sweden": "🇸🇪",
    "Denmark": "🇩🇰",
    "Iceland": "🇮🇸",
    "Georgia": "🇬🇪",
    "Armenia": "🇦🇲",
}


def country_to_flag(country: str) -> str:
    return COUNTRY_FLAGS.get(country.strip(), "🌍")


def default_trip_name(country: str, year: int | None = None) -> str:
    return f"{country.strip()} {year or datetime.now().year}"


def parse_destination(destination: str) -> tuple[str, str]:
    if not destination:
        return "", ""
    parts = [part.strip() for part in destination.split(",") if part.strip()]
    if len(parts) >= 2:
        return parts[0], parts[-1]
    if len(parts) == 1:
        return parts[0], ""
    return "", ""

File: frontend/src/lib/countries.ts
export const COUNTRY_OPTIONS = [
  { name: "Japan", flag: "🇯🇵" },
  { name: "Italy", flag: "🇮🇹" },
  { name: "China", flag: "🇨🇳" },
  { name: "France", flag: "🇫🇷" },
  { name: "Spain", flag: "🇪🇸" },
  { name: "Germany", flag: "🇩🇪" },
  { name: "United Kingdom", flag: "🇬🇧" },
  { name: "United States", flag: "🇺🇸" },
  { name: "South Korea", flag: "🇰🇷" },
  { name: "Thailand", flag: "🇹🇭" },
  { name: "Vietnam", flag: "🇻🇳" },
  { name: "Indonesia", flag: "🇮🇩" },
  { name: "Turkey", flag: "🇹🇷" },
  { name: "Greece", flag: "🇬🇷" },
  { name: "Portugal", flag: "🇵🇹" },
  { name: "Netherlands", flag: "🇳🇱" },
  { name: "Switzerland", flag: "🇨🇭" },
  { name: "Austria", flag: "🇦🇹" },
  { name: "Czech Republic", flag: "🇨🇿" },
  { name: "Poland", flag: "🇵🇱" },
  { name: "Mexico", flag: "🇲🇽" },
  { name: "Brazil", flag: "🇧🇷" },
  { name: "Argentina", flag: "🇦🇷" },
  { name: "Morocco", flag: "🇲🇦" },
  { name: "Egypt", flag: "🇪🇬" },
  { name: "UAE", flag: "🇦🇪" },
  { name: "Singapore", flag: "🇸🇬" },
  { name: "India", flag: "🇮🇳" },
  { name: "Australia", flag: "🇦🇺" },
  { name: "Canada", flag: "🇨🇦" },
  { name: "Norway", flag: "🇳🇴" },
  { name: "Sweden", flag: "🇸🇪" },
  { name: "Denmark", flag: "🇩🇰" },
  { name: "Iceland", flag: "🇮🇸" },
  { name: "Georgia", flag: "🇬🇪" },
  { name: "Armenia", flag: "🇦🇲" },
] as const;

export function countryToFlag(country: string): string {
  return COUNTRY_OPTIONS.find((c) => c.name === country)?.flag ?? "🌍";
}

export function defaultTripName(country: string): string {
  return `${country.trim()} ${new Date().getFullYear()}`;
}
